### PR TITLE
Added ethereal to send mail for event cancellation

### PR DESCRIPTION
**Pull request by IshmeetSingh06**

### CODE REVIEW

File: app/services/events_service.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def cancel(id)
       self.errors = "Event not found"
     else
       self.errors = event.errors.full_messages unless event.update(cancelled: true)
+      send_cancelled_email(event) if self.errors.blank?
```
Comment (PramanSatya) [inline app/services/events_service.rb]: Simplify it:
```ruby
elsif event.update(cancelled: true)
  send_cancelled_email(event)
else
  self.errors = event.errors.full_messages
end
```

File: app/services/events_service.rb [hunk 2]
```diff
@@ -48,4 +49,10 @@ def list_registrations(id)
     event = Event.find_by(id: id)
     event ? event.registrations : self.errors = "Event not found"
   end
+
+  private def send_cancelled_email(event)
+    event.attendees.each do |user|
+      EventMailer.cancelled(user: user, event: event).deliver_now
```
Comment (PramanSatya) [inline app/services/events_service.rb]: Can we pass all the users in bcc without looping on each attendees?

File: spec/mailers/event_mailer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  describe 'event_cancellation_email' do
+    it 'renders the headers' do
+      expect(mail.subject).to eq('Event Cancelled')
+      expect(mail.to).to eq([user.email])
+      expect(mail.from).to eq(['from@example.com'])
+    end
+
+    it 'renders the body' do
+      expect(mail.body.encoded).to match(user.first_name)
```
Comment (PramanSatya) [inline spec/mailers/event_mailer_spec.rb]: You need to trigger the mail and check for increase in mailers count, here you are not triggering the mail anywhere

File: app/services/events_service.rb [hunk 4]
```diff
@@ -31,8 +31,10 @@ def cancel(id)
     event = Event.find_by(id: id)
     if event.blank?
       self.errors = "Event not found"
+    elsif event.update(cancelled: true)
+      EventMailer.cancelled(event: event).deliver_now
```
Comment (gagan93jtg) [inline app/services/events_service.rb]: should this be part of model callback, when `cancelled` is set to true?

File: spec/mailers/event_mailer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+RSpec.describe EventMailer, type: :mailer do
+  let(:admin) { FactoryBot.create(:user, role: 'admin') }
+  let(:event) { FactoryBot.create(:event) }
+  let(:mail) { EventMailer.cancelled(event: event) }
+
+  before(:each) { FactoryBot.create_list(:registration, 5, event: event) }
+
+  describe 'cancellation_email' do
+    it 'sends the email' do
+      debugger
```
Comment (gagan93jtg) [inline spec/mailers/event_mailer_spec.rb]: ??

File: app/mailers/event_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class EventMailer < ApplicationMailer
+  def cancelled(params)
```
Comment (gagan93jtg) [inline app/mailers/event_mailer.rb]: Can we just send the `event` object here as there is nothing else? `params` looks like a lot :)

**Review by gagan93jtg — CHANGES_REQUESTED** [review]

**Review by PramanSatya — APPROVED** [review]

**Pull request merged**